File: cronos-frontend/src/scripts/app/services/entity/usuarioService.js
angular.module('cronos')
    .factory('usuarioService', ['EntityService', function usuarioServiceFactory(EntityService) {
    'use strict';

    var UsuarioService = $class(EntityService, {

        constructor: function() {
            EntityService.call(this, {
                type: 'usuario'
            });
        },

        getAll: function() {
            return this._get('/all');
        },
        _findByGrupoTrabalho: function(id) {
            return this._get('/findByGrupoTrabalho/' + id);
        },
        _salvarUsuario: function(dto) {
            return this._post('/salvarUsuario', dto);
        },
        _removerUsuario: function(usuario) {
            return this._post('/removerUsuario', usuario);
        },
        getByCpf: function(cpf) {
            return this._get('/findByCpf?cpf=' + cpf);
        },
        /*
         * {
         * id,
         * usuarioAtualizacao
         * }
         **/
        aceiteTermo: function(aceiteTermoDTO) {
            return this._post('/aceiteTermo', aceiteTermoDTO);
        },
        _mudaPermissoes :function(usuarioDTO) {
            return this._post('/mudaPermissoes', usuarioDTO);
        }
    });

    return new UsuarioService();
}]);

File: cronos-service/src/main/java/br/jus/tjrs/cronos/app/model/service/UsuarioService.java
package br.jus.tjrs.cronos.app.model.service;

import java.util.List;

import javax.ejb.Local;

import br.jus.tjrs.cronos.app.model.dto.AceiteTermoDTO;
import br.jus.tjrs.cronos.app.model.dto.UsuarioDTO;
import br.jus.tjrs.cronos.app.model.entity.UsuarioEntity;
import br.jus.tjrs.cronos.core.CronosException;
import br.jus.tjrs.cronos.core.model.service.EntityService;
import br.jus.tjrs.cronos.core.rest.auth.AuthenticatedUserInfo;

@Local
public interface UsuarioService extends EntityService<Long, UsuarioEntity>
{
   UsuarioEntity findByCpf(String cpf) throws CronosException;

   UsuarioEntity createFirstUserMagistrate(AuthenticatedUserInfo userInfo) throws CronosException;

   UsuarioEntity findByLogin(String usuario) throws CronosException;

   UsuarioEntity aceiteTermo(AceiteTermoDTO aceite) throws CronosException;

   UsuarioDTO salvarUsuario(UsuarioDTO usuario) throws CronosException;

   List<UsuarioDTO> findByGrupoTrabalho(Long idGrupoTrabalho) throws CronosException;

   boolean removerUsuario(UsuarioDTO usuario) throws CronosException;

   UsuarioDTO mudaPermissoes(UsuarioDTO usuario) throws CronosException;
}
